### ISSUE TITLE
Can't show player on-top using hotkey

### ISSUE THREAD

**Issue by Solarunit** [DeaDBeeF-Player/deadbeef#1987]
With assigned global hotkey (show/hide player window) player can't be maximized on-top. 

![db](https://user-images.githubusercontent.com/20891064/37869824-19d963c4-2fd0-11e8-841d-085bc260a9fb.gif)

Deadbeef version: 0.7.2
OS: Solus 3.9 Budgie 10.4

**Comment by Alexey-Yakovenko** [comment]
I'm pretty sure that GNOME would prevent this issue from being fixed. It's most likely holds the current window in focus, while any key is pressed, or something like that.

**Comment by OVlasiuk** [comment]
Cinnamon (which is also based on Gnome 3) has a setting precisely to this effect: "Prevent focus stealing", disabling which fixes the same issue there. One would expect that Budgie has a similar setting as well.

**Comment by Alexey-Yakovenko** [comment]
Reopening to try an idea:

The action implementation can defer putting window in focus by say 200ms, this may trick focus stealing preventor

**Comment by AngryPhantom** [comment]
I can confirm this too. If I add a Hotkey in Preferences/Hotkeys the player's main window is never in the foreground for some reason.

**Comment by Alexey-Yakovenko** [comment]
I added a potential fix for this. I don't have a way of testing it right now, would be nice if somebody try it out.

**Comment by Alexey-Yakovenko** [comment]
To try it out, as usual, use the latest build off master branch: https://sourceforge.net/projects/deadbeef/files/travis/linux/master/deadbeef-static_devel-1_x86_64.tar.bz2/download

**Comment by AngryPhantom** [comment]
Now it behaves a bit differently. It still doesn't go to the foreground when other apps/windows are active/open as before. But if I click on the desktop (thus making other apps inactive) and then invoke Deadbeef with the hotkey it becomes **active**, but only once. If I press the hotkey again to hide it and once again to show - it behaves as before...

**Comment by Alexey-Yakovenko** [comment]
ok.. I increased delay to 500ms, let's see if it changes anything.. there's some hope, but if this doesn't help -- I'm quite out of ideas.

**Comment by AngryPhantom** [comment]
But it is always active if it is the only window on the desktop (while others are minimized).

**Comment by Alexey-Yakovenko** [comment]
Also, I found an extension that disables GNOME focus stealing prevention: https://extensions.gnome.org/extension/2182/noannoyance/

I don't know if it works though. Apparently many people consider this a GNOME bug, but GNOME devs consider this a feature, and won't fix it.

**Comment by AngryPhantom** [comment]
Tried the release version. Actually it behaves the same as the _devel_ one.

**Comment by AngryPhantom** [comment]
XFCE has some Focus prevention settings, but they are useless in this case (tried every variant there).

**Comment by AngryPhantom** [comment]
Oops, I'm really sorry. One should really **DISABLE** Focus prevention. At least, in XFCE and use the "standard ICCCM hint" in the settings (afaik, it is enabled by default). So, I guess, we can close the issue. There's nothing to do here. And, as you said, GNOME users should use the appropriate extension.

**Comment by Alexey-Yakovenko** [comment]
Let's see if any GNOME user can confirm that the proposed fix didn't do anything useful, then I'd revert it.

**Comment by AngryPhantom** [comment]
Well, in Debian with MATE it's impossible to fix the issue because it has no such option/function as Focus Stealing Prevention at all. I guess it's hardcoded somewhere in Gtk :(

**Comment by AngryPhantom** [comment]
And what's really strange is that when I set Deadbeef window to be "Always On Top" the option resets as if I didn't select it.

**Comment by Alexey-Yakovenko** [comment]
I reverted the changes since they don't do anything useful. The problem remains unfixable on application side.

**Comment by Alexey-Yakovenko** [comment]
@AngryPhantom 
> And what's really strange is that when I set Deadbeef window to be "Always On Top" the option resets as if I didn't select it.

Unrelated to this issue, and unrelated to Deadbeef.
Deadbeef doesn't do anything with "always on top" -- it's a WM feature, and whatever happens is caused by the WM.
